### PR TITLE
Allow injection of cards

### PR DESCRIPTION
**Pull request by benjick**
WIP

### CODE REVIEW

File: components/ProductCard.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <template>
   <div class="card" data-testid="productCard">
     <div class="product-labels">
-      <div v-for="label in product.productLabels" :key="label.code">
+      <div v-for="label in product.productLabels" :key="label.sortey">
```
Comment (coopjakob) [inline components/ProductCard.vue]: Sortey, what is that?

Comment (benjick) [inline components/ProductCard.vue]:
```suggestion
      <div v-for="label in product.productLabels" :key="label.sortKey">
```

Comment (benjick) [reply components/ProductCard.vue]: Whoops, my `k`-key is a bit broken, thanks!

Comment (benjick) [reply components/ProductCard.vue]: Changed to `sortKey`

Comment (coopjakob) [reply components/ProductCard.vue]: Ah, and `product` is more like a card (change name?), and all cards don't have a product.code and this is why each card have an sortKey?

File: components/ProductMatrix.vue [hunk 2]
```diff
@@ -2,19 +2,23 @@
   <div>
     <div v-if="error">{{ error }}</div>
     <div class="product-matrix">
-      <ProductCard
-        v-for="product in productList"
-        :key="product.code"
-        :product="product"
-      />
+      <div v-for="product in cards" :key="product.sortKey">
```
Comment (coopjakob) [inline components/ProductMatrix.vue]: `product` is more like a card (change name?), and all cards don't have a `product.code` and this is why each card have an `sortKey`?

Comment (coopjakob) [inline components/ProductMatrix.vue]: Should we rename it to `v-for="card in cards"`?

Comment (benjick) [reply components/ProductMatrix.vue]: @coopjakob exactly! And I also needed a way to be able to attach a card before or after another card. See this one:

```
function addSortKey(card: any) {
  if (card.before) {
    card.sortKey = `${card.code}_order1`
  } else if (card.after) {
    card.sortKey = `${card.code}_order3`
  } else {
    card.sortKey = `${card.code}_order2`
  }
  return card
}
```

Comment (benjick) [reply components/ProductMatrix.vue]: @coopjakob great minds think alike! 🤝

File: store/cards.ts [hunk 3]
```diff
@@ -34,16 +56,28 @@ function bySortKey(a: Card, b: Card) {
...
+        // ['2317438900009', ''],
+        ['', '32484'], // Frukt och grönt
+        ['', '32484'], // Frukt och grönt
+        ['', '32484'], // Frukt och grönt
+        ['', '39033900'] // Vegetariskt
+      ].map(([product, category]) => {
+        const card: Card = {
+          name: 'Test',
+          sortKey: '',
+          id: uniqid(),
```
Comment (coopjakob) [inline store/cards.ts]: Why do we create a new ID with `uniqid`?
`product.code` exist(?) and new cards can also have a name that can be used as ID.

File: components/ProductMatrix.vue (hunk 2 above)
Comment (coopjakob) [reply components/ProductMatrix.vue]: I guess that you need to re-sort the array by sortKey after this?

Why not add it to the array at a specific position using splice or similar?
arr.splice(position, 0, object);

File: store/cards.ts (hunk 3 above)
Comment (benjick) [reply store/cards.ts]: Oops, forgot to remove that from me trying stuff out

Comment (benjick) [reply store/cards.ts]: Thanks!

**Comment by benjick** [comment]
> Maybe we need documentation of Siblings and what it is and how it works.

A sibling is the product card that the extra card was attached to

> Seems similar to my term buddy.

That's a great term! 

> Do we need a function to set a variable? What am I missing? Do I need to learn about commit?

In vuex you call an action, this action might do `fetch(url)` and then finally `commit('myMutation', data)`. This will trigger the mutation called `myMutation` that might look like this:
```
myMutation(state, data) {
  state.something = data.bar
}
```
You only change data in mutations.

**Comment by coopjakob** [comment]
> > Do we need a function to set a variable? What am I missing? Do I need to learn about commit?
> 
> In vuex you call an action, this action might do `fetch(url)` and then finally `commit('myMutation', data)`. This will trigger the mutation called `myMutation` that might look like this:
> 
> ```
> myMutation(state, data) {
>   state.something = data.bar
> }
> ```
> 
> You only change data in mutations.

Ok, the commit can't call `state.something = data.bar` it can only call an function/action?

**Comment by benjick** [comment]
> Ok, the commit can't call state.something = data.bar it can only call an function/action?

Exactly, you call a mutation by name

**Pull request merged**